app/views/layouts/_adv_search_body.html.haml

```diff
@@ -79,7 +79,7 @@
             %label.control-label.col-md-5
               = _("Global search:")
             .col-md-6
-              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:expression][:selected][:typ] == "global"
+              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:search_type] == "global"
```

Comment:
For me everything works perfect with my solution and I think it is good that it depends on what is in @edit[:search_type]. Without my fix it can happen that user saves a filter with wrong type (especially when he creates more filters with different types and names). It has happened to me multiple times as I was trying to fix BZ related to global filters.
For example my checkbox was unchecked and it saved filter and it appeared under global filters which was wrong.

If I choose a filter which has type` 'default'` and if I edit it and hit Save, I can see the checkbox 'Global search' is unchecked, which is right because as BZ says, we don't wanna have more filters with the same name under Global filters. Normally, we can create/save only filters with type 'global' or 'nil' (then it is a filter under My Filters), not default filters. Default filters are default, you cannot edit them, just to create the new one with the same name - and then it has to be a filter which is not global.

If I choose filter which has type 'global' and if I edit it and hit Save, I can see the checkbox IS CHECKED which is right again and it says that we are editing the actual global filter, not creating the new one. If I unchecked the checkbox and hit Save, it would create a new filter with the same name under My Filters (if no another filter with the same name was there). I have tried it multiple times in many variations, combinations or if switching to other searches/pages and it still works for me.

The little problem is that we don't know if we are trying to edit global or default filter because both types are under 'Global Filters' in accordion. You can check type of selected filter in console. We cannot recognize these filters without any action.

"@edit[:search_type] contains the type of the edited expression or what the user last selected or what the last expression was " and I think that this is completely ok according to how Advanced search was designed:
https://github.com/ManageIQ/manageiq-ui-classic/issues/387